### ISSUE TITLE
The theme color reference is a bit of a mess

### ISSUE THREAD

**Issue by fabiospampinato** [Microsoft/vscode#64614]
I'm maintaining a [theme](https://marketplace.visualstudio.com/items?itemName=fabiospampinato.vscode-monokai-night) with the stated goal that it themes every single themeable color. While developing and maintaining it I'm finding a few issues with the whole [color reference](https://code.visualstudio.com/docs/getstarted/theme-color-reference) thing.

### Colors missing from the reference 

My theme is currently theming the following colors:

- breadcrumb.background
- editorGroup.background

They are not even mentioned in the reference, I guess I got them from a release note, who knows.

### Missing colors from release notes (maybe?)

In my theme's first release I've themed every single color, and I've always paid attention to the release notes for any added colors. Still, quite surprisingly to me, all these colors are now present in the reference but my theme is not customizing them:

- diffEditor.border
- diffEditor.insertedTextBorder
- diffEditor.removedTextBorder
- editor.findRangeHighlightBorder
- editorOverviewRuler.commonContentForeground
- editorOverviewRuler.rangeHighlightForeground
- editorOverviewRuler.selectionHighlightForeground
- editorRuler.foreground
- editorUnnecessaryCode.border
- editorUnnecessaryCode.opacity
- editorWidget.resizeBorder
- gitDecoration.addedResourceForeground
- gitDecoration.submoduleResourceForeground
- list.inactiveFocusBackground
- list.invalidItemForeground
- merge.border
- merge.commonContentBackground
- merge.commonHeaderBackground
- notification.background
- notification.buttonBackground
- notification.buttonForeground
- notification.buttonHoverBackground
- notification.errorBackground
- notification.errorForeground
- notification.foreground
- notification.infoBackground
- notification.infoForeground
- notification.warningBackground
- notification.warningForeground
- panel.dropBackground
- peekViewEditor.matchHighlightBorder
- statusBar.debuggingBorder
- statusBar.noFolderBorder
- tab.hoverBackground
- tab.hoverBorder
- tab.unfocusedHoverBackground
- tab.unfocusedHoverBorder

Maybe I did a very poor job on my first release and left out all of them, but I think it's more likely that they have been added over time but they have not been mentioned in the release notes. If that's the case that's a problem, we can't be expected to keep monitoring the reference for updates.

### Difficult to theme properties

Most properties are obvious, but some of them I didn't immediately understand what they were theming, like: 

- editor.focusedStackFrameHighlightBackground
- editor.stackFrameHighlightBackground

Some more screenshots would be appreciated.

Also in order to theme some colors, like:

- notification.errorBackground
- notification.errorForeground
- terminal.ansiBlack
- terminal.ansiBlue

We need to see how they are actually being used, i.e. we need to trigger all kinds of custom notifications and have a script that uses all colors in the terminal. It might be good to develop some "official" scripts for this or at least link to third-party's scripts we could use.

### Incomplete colors support

The whole set of provided themeable colors seems incomplete to me. It seems to me almost like unless a particular color in a particular state is not themed explicitly by VSCode itself then that color isn't themeable at all by themes.

One quick example, tabs now have a top border, and they can be in one of these states: active, inactive, unfocused-active, unfocused-inactive.

If we want to customize the top border for active tabs we have tab.activeBorderTop, for unfocused-active tabs we have tab.unfocusedActiveBorderTop, and the other states just aren't themeable (or I guess those properties aren't mentioned in the reference?). 

### Poorly searchable/manipulable properties

Coming up with these lists of properties was a bit painful, it would be better if there was a JSON file or something that listed all of them, so that I could diff my theme's settings with it. Also seeing what values are assigned to those properties by default would be useful. Maybe this file exists already somewhere in the repo, if it does it should be mentioned in the reference as well.

**Comment by usernamehw** [comment]
> Colors missing from the reference

Should probably go there https://github.com/Microsoft/vscode-docs/issues

> it would be better if there was a JSON file or something

https://github.com/Microsoft/vscode-docs/issues/1588
TLDR: `Developer: Generate Color Theme From Current Settings` should return JSON with all `colors`

> Also in order to theme some colors, like:
> 
> notification.errorBackground
> notification.errorForeground

https://marketplace.visualstudio.com/items?itemName=svipben.notification-tester

**Comment by fabiospampinato** [comment]
Another missing color: `tab.hoverForeground`, there doesn't seem to be any good reason to me for not adding this one, the whole thing is just very incomplete.

**Comment by fabiospampinato** [comment]
Another property difficult to customize: `tab.hoverBorder`. This one should be easy, but I'm not seeing any changes at all 🤷‍♂️.

**Comment by fabiospampinato** [comment]
Another weird one: `tab.hoverBackground` (there might be a problem when all properties I try to customize have issues...). If one assigns a color to it then all tabs, even the active one will use that color when they are hovered. Now, one may think that by default this is assigned to a specific color, like (all?) the other colors. But not, if this property is not customized then tabs just don't change their background color on hover, totally undocumented, totally incomplete, there should be analogous properties for when the tab is also active etc.

**Comment by fabiospampinato** [comment]
>  totally incomplete, there should be analogous properties for when the tab is also active etc.

Weirdly there are properties for customizing unfocused tabs (`unfocusedHoverBackground`), which I would guess most themes won't care about using, but an `activeHoverBackground`, which actually makes sense, is not present.

**Comment by fabiospampinato** [comment]
Weirdness #∞+1: I tried setting `editorUnnecessaryCode.opacity` to 1, but the schema expects a string. Fine, I wronte `"1"` then, but the schema is actually excepting a color string, which doesn't make sense for the opacity.

**Comment by fabiospampinato** [comment]
> editorUnnecessaryCode.opacity

I'd also like to know that the default value for this is. Unfortunately autocompleting it inserts the value `"#ff0000"`...

**Comment by fabiospampinato** [comment]
> editorUnnecessaryCode.opacity

This doesn't seem to be working at all, whatever I set it to the opacity becomes 1, maybe the buggy schema thinks I'm setting an invalid value (I tried setting `0.5`, `.5` and `askjdn`) and defaults to "1"? 🤷‍♂️

**Comment by fabiospampinato** [comment]
> list.inactiveFocusBackground

The reference says "An active list has keyboard focus, an inactive does not. Currently only supported in lists." I haven't been able to see how this color is being used... I selected a list item via the keyboard, then switched the focus to an editor, nothing.

> list.invalidItemForeground

The reference mentions "for example an unresolved root in explorer.", so I added a folder to the workspace and then deleted it, without removing it from the workspace, nothing happened. Apparently one has to first save the workspace, then reopen it again, weird.

**Comment by fabiospampinato** [comment]
Setting a transparent (`#00000000`) border as the value of `editor.findRangeHighlightBorder` strangely causes kind of a border to appear, I'm not quite sure why:

<img width="1039" alt="screen shot 2018-12-07 at 21 52 54" src="https://user-images.githubusercontent.com/1812093/49672295-edecda00-fa6a-11e8-8db1-f23ee7f8a828.png">

![image](https://user-images.githubusercontent.com/1812093/49672289-e2011800-fa6a-11e8-9b0d-4ded47751458.png)

**Comment by fabiospampinato** [comment]
When setting the following properties:

- diffEditor.insertedTextBorder
- diffEditor.removedTextBorder

The added border is not added properly, since the whole highlight becomes misaligned. Here there's a before/after comparison:

<img width="908" alt="screen shot 2018-12-07 at 22 19 01" src="https://user-images.githubusercontent.com/1812093/49673249-24782400-fa6e-11e8-8ca0-01f3727772cd.png">
